app/form_models/jobseekers/job_application/details/qualifications/secondary/common_form.rb

```diff
@@ -28,6 +28,15 @@ def read_attribute_for_validation(attr)
       super unless /^qualification_results_attributes_/.match?(attr)
     end
 
+    def highest_present_result_index
```

Comment:
We can make this a bit terser with:

```ruby
@qualification_results.map.with_index { |result, index| index unless result.empty? }.compact.last || 0
```

You can also drop the `@` because of the attr_accessor.

Reply:
Also let's use conditional assignment `||=` here, as the method will be called once for each qualification.